src/SocketBuffer.c

```diff
@@ -132,7 +132,25 @@ void SocketBuffer_terminate(void)
  */
 void SocketBuffer_cleanup(int socket)
 {
+	pending_writes* pw;
+
 	FUNC_ENTRY;
+	/* Clean up the write data queued for the socket */
+	do
+	{
+		int i;
+
+		pw = SocketBuffer_getWrite(socket);
+		if (pw)
+		{
+			for (i = 0; i < pw->count; i++)
+			{
+				if (pw->frees[i])
+					free(pw->iovecs[i].iov_base);
+			}
+		}
```

Comment:
Does the function `MQTTAsync_writeComplete()` needs to be called for each freed pending_writes message,
in order to dealocate the corresponding write message command?